### ISSUE TITLE
Repeated Dummy in integrals incorrectly become distinct

### ISSUE THREAD

**Issue by latot** [cbm755/octsympy#679]
Hi, i found this interesting things, and a very big problem, actually we use the ```Dummy``` function to solve or pass some functions, and avoid use user env vars, now the problem is octsympy save the result of the expression in a char, and then is revalued, now the problem is there, when octsympy re-eval the expression Sympy detect the new Dummy as differents, and with it we can't eval or continue using some expressions................................

follow this in python:
```
>>> alpha = Dummy("alpha")
>>> i = Integral(1/sqrt(1 - sin(alpha)**2), (alpha, 0, pi/2))
>>> N(i)
0.e+2
>>> i = S(srepr(i))
>>> N(i)
Integral(1/sqrt(-sin(_alpha)**2 + 1), (_alpha, 0, pi/2))
```
The ```S(srep(something))``` is what octsympy does, so the problem with this is the eval and similar of expressions, like integrals, subs, diffs, all in what we can need replace a tmp var with it, the cases in what this don't affect is when we replace the ```Dummy``` for example with other expression, but if we need use it as a tmp var this is problem.

Example:
https://github.com/latot/octsympy/blob/096142db6ca2ca37b142ac2c9bca3909e4856964/inst/%40sym/ellipticE.m

If we use Dummy instead of alpha we can't eval the expression in octave.

Thx. Cya.

**Comment by cbm755** [comment]
Saw your upstream bug; linking here: https://github.com/sympy/sympy/issues/11847

**Comment by latot** [comment]
ups!, Thx!

**Comment by cbm755** [comment]
This is fixed upstream in upcoming Sympy 1.1.
